### DISCUSSION TITLE
Synchronization of issues

### DISCUSSION THREAD

**Discussion by VedankNaik** [iTwin/itwinjs-core#5186] [Q&A] [answered]
Hello,

I have implemented create issue functionality. I am able to create and list issues in my app.  But issues created using app are not listed on imodelweb. Is there anything that needs to be done so that issues are displayed at both places? 

Thanks.

**Comment by arnab-git-hub** [comment]
Are you trying to visualize the issues in the iModel viewer?

**Reply by VedankNaik** [reply]
Using widget as shown in [sandbox sample](https://www.itwinjs.org/sandboxes/iTwinPlatform/Issues)

**Reply by arnab-git-hub** [reply]
Can we set up a call to understand and try to resolve the issue?

**Reply by VedankNaik** [reply]
Yes. Please suggest suitable time and contact details.

**Reply by arnab-git-hub** [reply]
I have emailed you regarding the timing.

**Comment by pratikshan85** [comment, marked as answer]
@VedankNaik 

Please find the below steps to create and assign the issue to iModel.
Once we create an issue using API it will create an issue under the iTwin, it won’t associate the created issue to the iModel.

Step 1 . Create an Issue using API POST https://api.bentley.com/issues/
Step 2. Update the newly created issue to associate with an iModel   PATCH [https://api.bentley.com/issues/{id}](https://api.bentley.com/issues/%7bid%7d)

Body: {
    
    "properties": {
            "_ContainerId": "5040a636-f17f-4cc3-9cd1-229c533560db"
     }
}
 ContainerId is the iModelId.

Once we associate an issue with an iModel then only we can see the issues in the iModel web viewer.

In your local application, all the issues are getting displayed because Get Issue API is getting called, and as the response, it is giving us all the issues as it is under the same iTwin Irrespective of whether the issues are assigned to an iModel or not.

Please let me know if you need anything more.

Thanks & Regards
Pratikshan

**Reply by VedankNaik** [reply]
Thanks @pratikshan85

I am able to see issues in iModel web now, but thumbnails of the issues are not displaying in iModel web (although I can see thumbnails in the local application). The attachments list on iModel web shows that there are attachments, but the thumbnail placeholder is blank.

Additionally, could you please explain how to handle [Secondary Assignee](https://developer.bentley.com/apis/issues-v1/operations/create-issue/#secondary-assignee)? I created data similar to what is shown in sample requests, but the API calls are failing for the same.

**Reply by pratikshan85** [reply]
@VedankNaik Can you please provide the exact error details when you use **Secondary Assignee** in the post data to create the issue ?

**Reply by RameshwarShindeCctech** [reply]
Hi @pratikshan85,

As at my end too facing the same issue. Attaching a snap below for such.

![image](https://user-images.githubusercontent.com/111970621/228537170-58da7a4b-c172-40d8-8ff3-107a431aa008.png)

**Reply by pratikshan85** [reply]
[@Rameshwar](https://github.com/RameshwarShindeCctech) Can you please provide me the payload or the post data that you are passing while creating the issue?

**Reply by RameshwarShindeCctech** [reply]
All right. Here is the payload I'm sending.

![image](https://user-images.githubusercontent.com/111970621/228753057-ec3c8a36-fd99-4938-9e16-0d90fb9ad4c6.png)

**Reply by pratikshan85** [reply]
[@Rameshwar](https://github.com/RameshwarShindeCctech) Can you please give me the data in JSON format ?

**Reply by RameshwarShindeCctech** [reply]
{
  "formId": "9iu1f_82dUWnE9i7JQDtWxt0AGQpNX5Jj-R8I4zTJ-g",
  "subject": "Material Damage",
  "description": "Needs Repair.",
  "dueDate": "2023-03-31",
  "status": "Open",
  "assignees": [
    {
      "displayName": "CCtech CCtech",
      "id": "a4246ef0-c1a5-453f-9f63-1fed75726331"
    }
  ],
  "modelPin": {
    "location": {
      "x": 10.581614133114039,
      "y": 11.071581681678271,
      "z": 1.8669393452231315
    }
  },
  "modelView": {
    "iModelJsView": "",
    "cameraView": {
      "viewPoint": {
        "x": 4.531276830977632,
        "y": -29.166823663984474,
        "z": 23.137049280692388
      },
      "direction": {
        "x": 26.40205719699299,
        "y": 47.4564875143226,
        "z": -20.933804504564343
      },
      "up": {
        "x": 0.17486417438931334,
        "y": 0.31431033751240867,
        "z": 0.9330763807148801
      },
      "fieldOfView": 62.282886528241136
    }
  },
  "boundingBox": {
    "lowerLeftPoint3D": {
      "x": 0,
      "y": 6.661338147750939e-16,
      "z": -5.018121667163394
    },
    "upperRightPoint3D": {
      "x": 10.266799652529027,
      "y": 3.3613604599143283,
      "z": 0
    }
  },
  "properties": {
    "_ContainerId": "6dc59522-9d80-4a07-bba4-0073c856c8c9"
  }
}

**Reply by pratikshan85** [reply]
@RameshwarShindeCctech @VedankNaik 
Hello,

This API call failed because you don't have the Assign permission for the form you're creating, so you're not allowed to set assignee or assignees.  

Now, to get permission for setting the assignee or assignees you need to have the 
**"Forms_AssignAccess"** permission.

Please refer to the below link for more details about the Assign permission (Please check the permission header).

https://developer.bentley.com/apis/issues-v1/operations/modify-issue/


To set the Permission for a member, please visit the below link.

https://developer.bentley.com/apis/access-control/operations/create-itwin-role/

Please let me know of any further details.

**Reply by VedankNaik** [reply]
Hello @pratikshan85,

We checked links you shared above, to create a role it is mention we need `administration_manage_roles` permission. How/where do we assign this permission?

**Reply by pratikshan85** [reply]
@VedankNaik 
Yes, you need to have **administration_manage_roles** & **Forms_AssignAccess** roles both.
Please check the below steps to assign permission.
1. Create a role. 
2. Update the role with permission **Forms_AssignAccess**.
3. Update iTwin member with the newly created role.

All of the above steps can be achieved by calling API which is given in the below links.
1. https://developer.bentley.com/apis/access-control/operations/create-itwin-role/
2. https://developer.bentley.com/apis/access-control/operations/update-itwin-role/
3. https://developer.bentley.com/apis/access-control/operations/update-itwin-member/

**Reply by VedankNaik** [reply]
@pratikshan85,
I tried running create role API, it gives me insufficient permission error.
![image](https://user-images.githubusercontent.com/47364632/229509070-13d6589c-df06-48c0-95d0-9b56ecba221f.png)

**Reply by pratikshan85** [reply]
@VedankNaik , We need to have a quick discussion on this. I am sending you email regarding this.
